vocab.py

```diff
@@ -46,6 +48,21 @@ def __str__(self) -> str:
         return str(self.name)
 
 
+# Exception(s)
+
+class Error(Exception):
+    '''Base class for exceptions for the project'''
+
+
+class JobError(Error):
+    '''Exception raised for errors that should prompt a job cancellation'''
```

Comment:
Does `JobError` need to be based on `Error`, also defined here?  Isn't there an existing Python error that could be used or just base it on `Exception` directly?

Also, isn't `Error` ambiguous?  Couldn't there be conflicts with `Error` classes from other modules used by this application?  I'm thinking it should be called `CourseInventoryBaseError`, if the class is really necessary.

Reply:
Yeah seems like `JobError` could just extend `Exception` unless we were doing more with Error. We could probably also call it `JobException(Exception):`

Reply:
I was following what the docs (https://docs.python.org/3/tutorial/errors.html#user-defined-exceptions) describe, since it was presented as a best practice. I can rename or remove it if you don’t think it’s important.

Reply:
There is no base Python `Error` class that I could find, just `Exception`.

Reply:
`SomethingError` seems to be the standard naming pattern for user-defined Exceptions, and Exceptions in general (e.g. `KeyError`).

Reply:
I didn't want to wait any longer for more direction, so I changed the base class to `CourseInventoryBaseError`, as @lsloan suggested.

Reply:
@ssciolla, I'm sorry if you were frustrated by waiting for responses.  After I gave my review for the PR, I became engrossed in some of the recent MiVideo questions and I found a bug to be corrected.  I didn't notice you'd been waiting for more feedback.

I had hoped that my approval of your PR and the way I'd worded my comments you'd understand that I didn't mean them to block the merge.  I just wanted to make sure a potential problem had enough attention.

FWIW, yes, I saw that you'd cited the exception tutorial.  It's a good resource and I'm glad you mentioned it. However, I think if everybody follows it and names their base exception `Error`, there's even more possibility for ambiguity and conflict.  I wish they would address that in the tutorial.  Either to say that base exception classes should have more explicit names or to confirm that I'm wrong about ambiguity and conflict.  (Maybe I can open an issue to request the tutorial be updated.)